src/arrayMethodJoin.js

```diff
@@ -3,9 +3,49 @@
 /**
  * Implement method join
  */
+
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      switch (this.length - 1) {
+        case i:
+          if (this[i] !== undefined) {
+            str += this[i];
+          }
+
+          continue;
+      }
+
+      switch (this[i]) {
+        case null:
+          str += separator;
+
+          continue;
+
+        case undefined:
+          str += separator;
+
+          continue;
+      }
+
+      switch (separator) {
+        case undefined:
+          str += this[i] + ',';
+
+          continue;
+
+        case null:
+          str += this[i] + 'null';
+
+          continue;
+      }
+
+      str += this[i] + separator;
+    }
```

Comment:
Looks too complicated, actually we only need two checks to complete this task:

1. check if this[i] is not null and is not undefined then we can add this[i] to our str
2. check if current index is less than the last index so we can add separator to our str

And thats it